### PR TITLE
honor no-rtti setting

### PR DESCRIPTION
**Pull request by yulin-li**
The `TYPENAME` method should honor the `no-rtti` build flag when using GCC building, just like win32 does.

### CODE REVIEW

**Review by maxgolov — APPROVED** [review]

**Comment by yulin-li** [comment]
@maxgolov thx for the reviewing, could you also help to merge this PR? I have no write access to this repo...

**Comment by maxgolov** [comment]
> @maxgolov thx for the reviewing, could you also help to merge this PR? I have no write access to this repo...

I'd like a second pair of eyes to take a look at it.

**Comment by yulin-li** [comment]
> > @maxgolov thx for the reviewing, could you also help to merge this PR? I have no write access to this repo...
> 
> I'd like a second pair of eyes to take a look at it.

Sure, that's ok

**Review by reyang — APPROVED** [review]

**Pull request merged**
